File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/viewmodel/ClienteViewModel.java
package com.example.agnciadeturismo.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.agnciadeturismo.data.repository.ClienteRepositoryTask;

public class ClienteViewModel extends ViewModel {

    private MutableLiveData<Boolean> mCadastrado = new MutableLiveData<>();
    public LiveData cadastrado = mCadastrado;

    ClienteRepositoryTask clienteRepositoryTask = new ClienteRepositoryTask();
    public static String cadastradoSucesso;
    boolean postCliente;

    public void modificarCliente(boolean alterar, String nome, String email, String cpf, String rg, String telefone, String senha, String img){
        new Thread(new Runnable() {
            @Override
            public void run() {
                if(!alterar){
                    inserirCliente(nome, email, cpf, rg, telefone, senha, img);
                }else{
                    updateCliente(nome, email, cpf, rg, telefone, senha, img);
                }
            }
        }).start();
    }

    private void inserirCliente(String nome, String email, String cpf, String rg, String telefone, String senha, String img) {
        postCliente = clienteRepositoryTask.inserirCliente(nome, email, cpf, rg, telefone, senha, img);
        cadastradoSucesso = "Cadastrado com Sucesso!!";
        mCadastrado.postValue(postCliente);
    }

    private void updateCliente(String nome, String email, String cpf, String rg, String telefone, String senha, String img) {
        postCliente = clienteRepositoryTask.alterarCliente(nome, email, cpf, rg, telefone, senha, img);
        cadastradoSucesso = "Alterado com Sucesso!!";
        mCadastrado.postValue(postCliente);
    }
}

File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/presenter/viewmodel/ClienteViewModel.kt
package com.example.agnciadeturismo.presenter.viewmodel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.agnciadeturismo.data.repository.CidadeRepositoryTask
import com.example.agnciadeturismo.data.repository.ClienteRepositoryTask

class ClienteViewModel(private val clienteRepositoryTask: ClienteRepositoryTask) : ViewModel() {

    private val TAG = "ClienteViewModel"
    private val mCadastrado = MutableLiveData<Boolean>()
    var cadastrado: LiveData<Boolean> = mCadastrado

    var postCliente = false
    var cadastradoSucesso: String? = null

    fun modificarCliente(alterar: Boolean, nome: String, email: String, cpf: String, rg: String, telefone: String, senha: String, img: String) {
        Thread {
            try{
                if (!alterar) inserirCliente(nome, email, cpf, rg, telefone, senha, img)
                else updateCliente(nome, email, cpf, rg, telefone, senha, img)
            }catch (ex : Exception){ Log.d(TAG, ex.toString())}
        }.start()
    }

    private fun inserirCliente(nome: String, email: String, cpf: String, rg: String, telefone: String, senha: String, img: String?) {
        postCliente = clienteRepositoryTask.inserirCliente(nome, email, cpf, rg, telefone, senha, img)
        cadastradoSucesso = "Cadastrado com Sucesso!!"
        mCadastrado.postValue(postCliente)
    }

    private fun updateCliente(nome: String, email: String, cpf: String, rg: String, telefone: String, senha: String, img: String) {
        postCliente = clienteRepositoryTask.alterarCliente(nome, email, cpf, rg, telefone, senha, img)
        cadastradoSucesso = "Alterado com Sucesso!!"
        mCadastrado.postValue(postCliente)
    }

    class ViewModelFactory(private val clienteRepositoryTask: ClienteRepositoryTask) : ViewModelProvider.Factory {
        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
            return modelClass.getConstructor(ClienteRepositoryTask::class.java)
                .newInstance(clienteRepositoryTask)
        }
    }
}